core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -282,6 +280,17 @@ final class ShuffleBlockFetcherIterator(
       }
     }
 
+    @inline def updateMergedReqsDuration(wasReqForMergedChunks: Boolean = false): Unit = {
+      if (remainingBlocks.isEmpty) {
+        val durationMs = TimeUnit.NANOSECONDS.toMillis(clock.nanoTime() - requestStartTime)
+        if (wasReqForMergedChunks) {
+          shuffleMetrics.incRemoteMergedReqsDuration(durationMs)
+        } else {
+          shuffleMetrics.incRemoteReqsDuration(durationMs)
+        }
```

Comment:
`shuffleLocalMetricsUpdate` and `shuffleRemoteMetricsUpdate` is treating merged fetches as a subset of all fetches - while here we are treating them seperately.
Any particular reason for this difference ?

Reply:
+1 on this point, the merged remote request duration should be counted into the total remote request duration, following the same logic for the local/remote bytes read.
@otterc WDTY?

Reply:
We added these metrics internally so we can easily figure out what is taking longer- the fetch of the merge chunks or the few unmerged blocks. If we include the duration of merged chunks in the total then we have to derive the value for un-merged blocks which is fine. It makes sense to be consistent to avoid confusion.